src/plot_utils.py

```diff
@@ -96,7 +102,18 @@ def heatmap(bdb, bql, **kwargs):
     """
     df = bqlu.cursor_to_df(bdb.execute(bql))
     df.fillna(0, inplace=True)
-    return zmatrix(df, **kwargs)
+    if selectors is None:
+        return zmatrix(df, **kwargs)
+    results = []
+    for n0selector in selectors:
+      n0selection = deps['name0'].map(n0selector)
+      for n1selector in selectors:
+        n1selection = deps['name1'].map(n1selector)
+        this_block = deps[n0selection & n1selection]
+        results.append(zmatrix(this_block, vmin=0, vmax=1))
+        plt.show()
+        plt.close('all')
```

Comment:
I agree with you both, and plan to remove it, just haven't gotten around to
it. Thanks for noticing!


Gregory A. Marton  617-858-0775
http://goo.gl/Ne09o
http://csail.mit.edu/~gremio


On Wed, Sep 30, 2015 at 10:06 PM, F Saad <notifications@github.com> wrote:

> In src/plot_utils.py
> <https://github.com/probcomp/bdbcontrib/pull/49#discussion_r40873585>:
>
> > @@ -96,7 +102,18 @@ def heatmap(bdb, bql, **kwargs):
> >      """
> >      df = bqlu.cursor_to_df(bdb.execute(bql))
> >      df.fillna(0, inplace=True)
> > -    return zmatrix(df, **kwargs)
> > +    if selectors is None:
> > +        return zmatrix(df, **kwargs)
> > +    results = []
> > +    for n0selector in selectors:
> > +      n0selection = deps['name0'].map(n0selector)
> > +      for n1selector in selectors:
> > +        n1selection = deps['name1'].map(n1selector)
> > +        this_block = deps[n0selection & n1selection]
> > +        results.append(zmatrix(this_block, vmin=0, vmax=1))
> > +        plt.show()
> > +        plt.close('all')
>
> Although here we close the plot?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/probcomp/bdbcontrib/pull/49/files#r40873585>.
>